lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -154,6 +154,21 @@ HRESULT JsrtDebugManager::DbgRegisterFunction(Js::ScriptContext* scriptContext,
         {
             utf8SourceInfo->SetDebugDocument(debugDocument);
         }
+
+        if (this->debugEventCallback != nullptr && !scriptContext->GetDebugContext()->GetIsReparsingSource())
+        {
+            JsrtDebugEventObject debugEventObject(scriptContext);
+            Js::DynamicObject* eventDataObject = debugEventObject.GetEventDataObject();
+            JsrtDebugUtils::AddSourceMetadataToObject(eventDataObject, utf8SourceInfo);
```

Comment:
>AddSourceMetadataToObject [](start = 28, length = 25)

When ever we use AddSourceMetadataToObject we also add Scriptid. Move the HasDebugDocument check and AddScriptIdToObject inside AddSourceMetadataToObject